### PR TITLE
Adding macros for wrapping around memory allocation code; fixing apparent leaks detected by these.

### PR DESCRIPTION
**Pull request by spkjp**

### CODE REVIEW

**Comment by spkjp** [comment]
I had the wrappers pointing to OPENSSL_malloc and such, but wasn't clear whether you'd
like them.  If you could keep using wrapper macros like these, I can always easily point them
to debugging memory functions...

File: libopenarc/arc.c [hunk 1]
```diff
@@ -2256,14 +2257,33 @@ arc_free(ARC_MESSAGE *msg)
...
+	h = msg->arc_sealhead;
+	while (h != NULL)
+	{
+		tmp = h->hdr_next;
+		ARC_free(h->hdr_text);
+		ARC_free(h);
+		h = tmp;
+	}
+
+if (msg->arc_hdrlist != NULL)
```
Comment (mskucherawy) [inline libopenarc/arc.c]: Fix indent, please.

File: libopenarc/arc-canon.c [hunk 2]
```diff
@@ -690,6 +690,12 @@ arc_canon_cleanup(ARC_MESSAGE *msg)
 	}
 
 	msg->arc_canonhead = NULL;
+
```
Comment (mskucherawy) [inline libopenarc/arc-canon.c]: Fix indent, please.

File: libopenarc/arc.c [hunk 3]
```diff
@@ -2256,14 +2257,33 @@ arc_free(ARC_MESSAGE *msg)
...
+{
+	ARC_free(msg->arc_hdrlist);
+}
+
+if (msg->arc_hdrbuf != NULL)
```
Comment (mskucherawy) [inline libopenarc/arc.c]: Fix indent, please.

**Comment by spkjp** [comment]
Hi Murray, sorry about that...  I think the indent may have gotten dropped
when I copied over from my build environment.  Should be fixed now.

**Comment by mskucherawy** [comment]
Do you know how I can merge this onto a branch other than "master"?  Or do you need to do a new PR?  I want to start putting these on "develop".

**Comment by spkjp** [comment]
I don't know...  Let me know if you need anything from me.
But, did you check out the memory leaks part?  They seem worthy of bringing
out.

.. Juan

**Comment by kurta** [comment]
On Fri, Dec 15, 2017 at 3:50 PM, Juan Altmayer Pizzorno <
notifications@github.com> wrote:

> I don't know... Let me know if you need anything from me.
> But, did you check out the memory leaks part? They seem worthy of bringing
> out.
>
> .. Juan


Juan,

We have heard a non-specific report of memory leaks from the person working
on implementing ARC for MessageSystem's Momentum/LUA too. If you have a
patch, I'm sure they would be interested.

--Kurt

**Comment by juh** [comment]
Juan,
for changing the branch the pull request is for, see https://help.github.com/articles/changing-the-base-branch-of-a-pull-request/

**Comment by spkjp** [comment]
Hi Kurt, I did share my patch with Elliot... I'm just coming back from leave, so I'll reach out to
him to see what's new.  Thanks!

.. Juan

File: libopenarc/arc-internal.h [hunk 4]
```diff
@@ -122,4 +122,13 @@ typedef struct arc_plist ARC_PLIST;
 struct arc_canon;
 typedef struct arc_canon ARC_CANON;
 
+/*
+**  memory allocation wrappers
+*/
+
+#define ARC_malloc  malloc
```
Comment (mskucherawy) [inline libopenarc/arc-internal.h]: I think macros should be all-caps.  The code looks good but I'd like to be consistent with current use throughout the code base.  Do you mind making that change and resubmitting one more time?

**Comment by mskucherawy** [comment]
I'll fix it myself.

**Pull request merged**